### DISCUSSION TITLE
求助：希望可以帮忙看一下如何设定tags和categories页面显示的文章数量？

### DISCUSSION THREAD

**Discussion by iqiyiqiyuvr** [CaiJimmy/hugo-theme-stack#900] [Q&A]
因为我的网站文章比较多，所以在生成静态页面时在生成分页的时候往往会消耗大量的时间。
于是我决定去除分页的功能，设置在主页显示最新的20篇文章。
尝试在layouts里的模板删除所有分页相关的代码后还是只能实现限制显示的文章数量，以及不显示分页页数的的导航。
于是换了个思路，config中设定paginate为一个很大的数字，这样分页只有1页了，然后再index.html模板中修改限制了只显示first 20个文章。这样总算基本实现原本的目标，大幅缩减了生成静态页面的时间了。
但是发现tags和categories页面还是一页显示全部的文章，于是尝试再list.html模板中subsections部分用同样的方式限定只显示first20篇文章，但怎么尝试都没有结果。
拜托大神百忙之中抽空指点一下，我该怎么限制tags和categories页面显示的文章数量呢？
这真的对我来讲很重要...因为文章数量太多，现在一页到底真的惨不忍睹，要么我就要每次等待很久生成静态文件，真的是两难...感恩！感谢！

**Comment by chuxinyuan** [comment]
[Hugo Stack主题装修笔记](https://thirdshire.com/post/hugo-stack-renovation/)这篇文章或许对你有帮助。
